src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -117,20 +217,20 @@ private function evaluateToken(JsonPathToken $token, mixed $value): array
 
     private function evaluateName(string $name, mixed $value): array
     {
-        if (!\is_array($value)) {
+        if (!$this->isArrayOrObject($value)) {
             return [];
         }
 
         if ('*' === $name) {
-            return array_values($value);
+            return $this->getValues($value);
```

Comment:
the getValues helper is not needed:
```suggestion
            return array_values((array) $value);
```